pkg/apis/vpc/v1alpha1/subnet_types.go

```diff
@@ -92,12 +95,12 @@ type SubnetList struct {
 }
 
 // SubnetDHCPConfig is DHCP configuration for Subnet.
-// +kubebuilder:validation:XValidation:rule="has(oldSelf.mode) || !has(self.mode) || self.mode=='DHCPDeactivated'", message="subnetDHCPConfig cannot switch from DHCPDeactivated to other modes"
+// +kubebuilder:validation:XValidation:rule="has(oldSelf.mode)==has(self.mode) || (has(oldSelf.mode) && !has(self.mode)  && oldSelf.mode=='DHCPDeactivated') || (!has(oldSelf.mode) && has(self.mode) && self.mode=='DHCPDeactivated')", message="subnetDHCPConfig mode can only switch between DHCPServer and DHCPRelay"
```

Comment:
do you test if it is allowed if !has(oldSelf.mode) && !has(self.mode)?

Reply:
Yes, it is allowed, has(oldSelf.mode)==has(self.mode) allows the case for both `!has(oldSelf.mode) && !has(self.mode)` and `has(oldSelf.mode) && has(self.mode)`